src/aggregation/contribution_tracker.c

```diff
@@ -185,34 +168,31 @@ void contribution_tracker_update_contribution(
   {
     /* AID does not exist in table. */
     state->aid_seed ^= aid;
-    entry->has_contribution = true;
     entry->contributor.contribution = contribution;
     state->distinct_contributors++;
 
     add_top_contributor(&state->contribution_descriptor,
                         &state->top_contributors,
                         entry->contributor);
-    return;
   }
-  else if (!entry->has_contribution)
-  {
-    /* AID exists but hasn't contributed yet. */
-    entry->has_contribution = true;
-    entry->contributor.contribution = contribution;
-
-    add_top_contributor(
-        &state->contribution_descriptor,
-        &state->top_contributors,
-        entry->contributor);
-    return;
+  else
+  { /* Aggregate new contribution. */
+    contribution_t combined = descriptor->contribution_combine(entry->contributor.contribution, contribution);
+    if (memcmp(&combined, &entry->contributor.contribution, sizeof(contribution_t)) != 0)
```

Comment:
What is the point of this check? It wasn't there before.

Reply:
look at the comment to it: https://github.com/diffix/pg_diffix/pull/352/commits/27e2e0e93521cc9daa4301dbbbec44f136106f9d

and then the follow-up comment in the other thread